src/Stratis.Bitcoin.Features.BlockStore/BlockStoreQueue.cs

```diff
@@ -65,17 +65,18 @@ public class BlockStoreQueue : IBlockStoreQueue
         private readonly AsyncQueue<ChainedHeaderBlock> blocksQueue;
 
         /// <summary>Batch of blocks which should be saved in the database.</summary>
-        /// <remarks>Write access should be protected by <see cref="getBlockLock"/>.</remarks>
+        /// <remarks>Write access should be protected by <see cref="blocksCacheLock"/>.</remarks>
         private readonly List<ChainedHeaderBlock> batch;
 
         /// <summary>Task that runs <see cref="DequeueBlocksContinuouslyAsync"/>.</summary>
         private Task dequeueLoopTask;
 
         /// <summary>Protects the batch from being modifying while <see cref="GetBlockAsync"/> method is using the batch.</summary>
-        private readonly object getBlockLock;
+        private readonly object blocksCacheLock;
 
         /// <summary>Represents all blocks currently in the queue & pending batch, so that <see cref="GetBlockAsync"/> is able to return a value directly after enqueuing.</summary>
-        private readonly Dictionary<uint256, ChainedHeaderBlock> pendingBlocks;
+        /// <remarks>Write access should be protected by <see cref="blocksCacheLock"/>.</remarks>
+        private readonly Dictionary<uint256, ChainedHeaderBlock> pendingBlocksCache;
```

Comment:
There was no need to change the names of `getBlockLock` and `pendingBlocks` it only bloats the PR (remember for next time).